### PR TITLE
Add cloudformation test case to PR analysis integration tests

### LINKED ISSUE #3345: Skip analysis for unchanged files when the analysis allows it
When the sensor context indicates that unchanged files can be skipped from analysis (in practice this is during PR analysis), the SonarJS analyzers should execute only the UCFG rule (from sonar-security frontend) for files marked as unchanged.

UCFG rule generates UCFG files required for security tainted analysis. The eslint-bridge loads this external rule dynamically according to the 'bundles' node command parameter and then the ESLint linter is created based on the request rules parameter.
Thus a second 'optimized' linter should be created in the initialization service with only this external rule (`ucfg`). And the JS/TS/YAML analysis services need a new flag triggering the use of the 'optimized' linter.

On the integration test side we need to validate that UCFG are properly generated for unchanged files. For that we need to add sonar-security plugin to the orchestrator and assert presence of either UCFG files (low level check) or security issues (then some security rule should be activated). Probably we should go for a first option as more reliable and controllable by us.

### PR DESCRIPTION
**Pull request by francoismora**
Fixes #3345

### CODE REVIEW

File: its/plugin/projects/pr-analysis-cloudformation-branch/file2.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+AWSTemplateFormatVersion: 2010-09-09
+Resources:
+  SomeLambdaFunction:
+    Type: "AWS::Serverless::Function"
+    Properties:
+      Runtime: "nodejs16.0"
+      InlineCode: >
+        console.log('Hello, World');;
```
Comment (vilchik-elena) [inline its/plugin/projects/pr-analysis-cloudformation-branch/file2.yaml:8]: let's add `// Noncompliant`

File: its/plugin/projects/pr-analysis-cloudformation-main/file1.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+AWSTemplateFormatVersion: 2010-09-09
```
Comment (vilchik-elena) [inline its/plugin/projects/pr-analysis-cloudformation-main/file1.yaml:1]: is it possible to add comment that this is same as in branch?

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/PRAnalysisTest.java [hunk 3]
```diff
@@ -98,6 +99,49 @@ void should_analyse_pull_requests(String language) {
...
+
+    try (var gitExecutor = testProject.createIn(projectPath)) {
+      gitExecutor.execute(git -> git.checkout().setName(Master.BRANCH));
+      BuildResultAssert.assertThat(scanWith(getMasterScannerIn(projectPath, projectKey)))
+        .logsAtLeastOnce("DEBUG: Analysis of unchanged files will not be skipped (current analysis requires all files to be analyzed)")
+        .logsOnce("DEBUG: Initializing linter \"default\"")
+        .doesNotLog("DEBUG: Initializing linter \"unchanged\"")
+        .logsOnce("file1.yaml\" with linterId \"default\"")
+        .logsOnce("file2.yaml\" with linterId \"default\"")
+        .logsTimes(String.format("INFO: %1$d/%1$d source files have been analyzed", Master.SOURCE_FILES), 2)
```
Comment (vilchik-elena) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/PRAnalysisTest.java:121]: why is it logged twice?

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/PRAnalysisTest.java [hunk 4]
```diff
@@ -98,6 +99,49 @@ void should_analyse_pull_requests(String language) {
...
+        .generatesUcfgFilesForAll(projectPath, "file2_SomeLambdaFunction_yaml", "file1_SomeLambdaFunction_yaml");
+      assertThat(getIssues(orchestrator, projectKey, null))
+        .hasSize(1)
+        .extracting(Issues.Issue::getComponent)
+        .contains(projectKey + ":file1.yaml");
```
Comment (vilchik-elena) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/PRAnalysisTest.java:126]: let's assert rule key

**Review by vilchik-elena — CHANGES_REQUESTED** [review]

File: its/plugin/projects/pr-analysis-cloudformation-branch/file2.yaml (hunk 1 above)
Comment (francoismora) [reply its/plugin/projects/pr-analysis-cloudformation-branch/file2.yaml]: Comment added

File: its/plugin/projects/pr-analysis-cloudformation-main/file1.yaml (hunk 2 above)
Comment (francoismora) [reply its/plugin/projects/pr-analysis-cloudformation-main/file1.yaml]: Comment added

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/PRAnalysisTest.java (hunk 3 above)
Comment (francoismora) [reply its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/PRAnalysisTest.java]: Two sensors log the message. As this is no longer relevant to the test case, I removed this assertion from the class test cases.

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/PRAnalysisTest.java (hunk 4 above)
Comment (francoismora) [reply its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/PRAnalysisTest.java]: Rule assertion added

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/PRAnalysisTest.java [hunk 5]
```diff
@@ -98,6 +99,49 @@ void should_analyse_pull_requests(String language) {
...
+        .logsOnce("DEBUG: Initializing linter \"unchanged\"")
+        .logsOnce("file1.yaml\" with linterId \"unchanged\"")
+        .logsOnce("file2.yaml\" with linterId \"default\"")
+        .logsTimes("DEBUG: Saving issue for rule no-extra-semi", PR.ANALYZER_REPORTED_ISSUES)
+        .logsTimes(String.format("INFO: %1$d/%1$d source files have been analyzed", PR.SOURCE_FILES), 2)
+        .generatesUcfgFilesForAll(projectPath, "file2_SomeLambdaFunction_yaml", "file1_SomeLambdaFunction_yaml");
+      assertThat(getIssues(orchestrator, projectKey, PR.BRANCH))
+        .hasSize(1)
+        .extracting(Issues.Issue::getComponent)
+        .contains(projectKey + ":file2.yaml");
```
[the remark this answers is not in the document]
Comment (francoismora) [reply its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/PRAnalysisTest.java]: Rule assertion added

**Review by vilchik-elena — APPROVED** [review]

**Pull request merged**